### ISSUE TITLE
D<0 mccompcurname issue McXtrace Test_CRL

### ISSUE THREAD

**Issue by willend** [McStasMcXtrace/McCode#844]
With todays master, one yields

<pre>
mxrun.pl --test Test_CRL
*** No directory given - placing data in Test_CRL_20190820_154127 ***
# mcxtrace self-test (mxrun.pl --test)
McXtrace version 1.5beta001 (Aug. 20, 2019)

# Installing 'selftest_VC7M' directory in /Users/pkwi/McXtrace-tst
# Copying instruments from /Applications/McXtrace-1.5beta001.app/Contents/Resources/mcxtrace/1.5beta001/examples/
# Counts:        1000000
# Output format: PGPLOT
# Start Date:    Tue Aug 20 15:41:27 2019
INSTRUMENT Test_CRL:
  Unit test instrument for various lens components.

Executing: mxrun.pl Test_CRL.instr LENS=0 L1=35.047 L2=17.523 -n 1000000 --format=PGPLOT -d Test_CRL_1
Warning: Component Origin=Progress_bar(string profile) definition parameter
         may be changed into a setting parameter to avoid
         warnings at compile time.
Warning: Component source=Source_flat(string spectrum_file) definition parameter
         may be changed into a setting parameter to avoid
         warnings at compile time.
Warning: Component crl2d=Lens_parab(string material_datafile) definition parameter
         may be changed into a setting parameter to avoid
         warnings at compile time.
Warning: Component crl1d=Lens_parab_Cyl(string material_datafile) definition parameter
         may be changed into a setting parameter to avoid
         warnings at compile time.
CFLAGS=
/Applications/McXtrace-1.5beta001.app/Contents/Resources/mcxtrace/1.5beta001/sources/Source_flat.comp:340:1: warning: 
      control reaches end of non-void function [-Wreturn-type]
}
^
Test_CRL.c:9402:16: warning: equality comparison with extraneous parentheses
      [-Wparentheses-equality]
if (( mcipLENS == 2 ))
      ~~~~~~~~~^~~~
Test_CRL.c:9402:16: note: remove extraneous parentheses around the comparison to
      silence this warning
if (( mcipLENS == 2 ))
    ~          ^    ~
Test_CRL.c:9402:16: note: use '=' to turn this equality comparison into an
      assignment
if (( mcipLENS == 2 ))
               ^~
               =
Test_CRL.c:9587:16: warning: equality comparison with extraneous parentheses
      [-Wparentheses-equality]
if (( mcipLENS == 1 ))
      ~~~~~~~~~^~~~
Test_CRL.c:9587:16: note: remove extraneous parentheses around the comparison to
      silence this warning
if (( mcipLENS == 1 ))
    ~          ^    ~
Test_CRL.c:9587:16: note: use '=' to turn this equality comparison into an
      assignment
if (( mcipLENS == 1 ))
               ^~
               =
Test_CRL.c:9786:16: warning: equality comparison with extraneous parentheses
      [-Wparentheses-equality]
if (( mcipLENS == 0 ))
      ~~~~~~~~~^~~~
Test_CRL.c:9786:16: note: remove extraneous parentheses around the comparison to
      silence this warning
if (( mcipLENS == 0 ))
    ~          ^    ~
Test_CRL.c:9786:16: note: use '=' to turn this equality comparison into an
      assignment
if (( mcipLENS == 0 ))
               ^~
               =
/Applications/McXtrace-1.5beta001.app/Contents/Resources/mcxtrace/1.5beta001/optics/Lens_parab_Cyl.comp:222:3: warning: 
      array index 6 is past the end of the array (which contains 6 elements)
      [-Warray-bounds]
  parab_Cyl.constants[6]=rough_xy;
  ^                   ~
/Applications/McXtrace-1.5beta001.app/Contents/Resources/mcxtrace/1.5beta001/optics/Lens_parab_Cyl.comp:56:7: note: 
      array 'constants' declared here
      double constants[6];
      ^
/Applications/McXtrace-1.5beta001.app/Contents/Resources/mcxtrace/1.5beta001/optics/Lens_parab_Cyl.comp:223:3: warning: 
      array index 7 is past the end of the array (which contains 6 elements)
      [-Warray-bounds]
  parab_Cyl.constants[7]=rough_z;
  ^                   ~
/Applications/McXtrace-1.5beta001.app/Contents/Resources/mcxtrace/1.5beta001/optics/Lens_parab_Cyl.comp:56:7: note: 
      array 'constants' declared here
      double constants[6];
      ^
6 warnings generated.
line 89: D<0 mccompcurname
line 89: D<0 mccompcurname
line 89: D<0 mccompcurname
line 89: D<0 mccompcurname
line 89: D<0 mccompcurname
line 89: D<0 mccompcurname
line 89: D<0 mccompcurname
line 89: D<0 mccompcurname
line 89: D<0 mccompcurname
line 89: D<0 mccompcurname
line 89: D<0 mccompcurname
line 89: D<0 mccompcurname
line 89: D<0 mccompcurname
line 89: D<0 mccompcurname
line 89: D<0 mccompcurname
line 89: D<0 mccompcurname
line 89: D<0 mccompcurname
line 89: D<0 mccompcurname
line 89: D<0 mccompcurname
line 89: D<0 mccompcurname
line 89: D<0 mccompcurname
line 89: D<0 mccompcurname
line 89: D<0 mccompcurname
line 89: D<0 mccompcurname
line 89: D<0 mccompcurname
line 89: D<0 mccompcurname
line 89: D<0 mccompcurname
line 89: D<0 mccompcurname
line 89: D<0 mccompcurname
line 89: D<0 mccompcurname
...
</pre>
and so forth

**Comment by ebknudsen** [comment]
I've done some testing on my machine (debian 10 buster)
Here it works with both clang 7 and gcc 8.

**Comment by willend** [comment]
Hmm, could be this is macOS specific then... Let's have a look together one of the next days

**Comment by ebknudsen** [comment]
still - this error message should really be removed in any case. It should be handled instead of just borking with a message.

**Comment by ebknudsen** [comment]
fixed by d6560b4e5248aacc5abefae67ac4f6ea76e90995

turns out to be an array bounds overrun
